backend/src/main/java/com/hufudb/onedb/backend/controller/UserController.java

```diff
@@ -87,6 +100,24 @@ void dropGlobalTable(@PathVariable String name) {
 
   @PostMapping("/user/query")
   PojoResultSet query(@RequestBody Request request) {
-    return PojoResultSet.fromResultSet(onedb.executeQuery(request.value));
+    Long id;
+    LocalDateTime submit = LocalDateTime.now();                           // get submit Time; Status = Submitted
+    synchronized (this) {
+      if (flag) {
+        maxId = sqlRecordService.getMaxId();
+        flag = false;
+      }
+      maxId = maxId + 1;
+      id = maxId;
+    }
+    sqlRecordService.submit(id, request.value, "qlh", submit);    // store into sqlRecord database
+    LocalDateTime start = LocalDateTime.now();                             // get start time
+    sqlRecordService.start(LocalDateTime.now(), id);
+    ResultSet rs = onedb.executeQuery(request.value);
+    LocalDateTime end = LocalDateTime.now();                               // get end time
+    String status = rs != null ? "Succeed" : "Failed";
```

Comment:
This step need two sql operation:
1. Insert
2. Update
3. The duration can be calculated when fetching the results.